copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopybookIOUtils.java

```diff
@@ -61,8 +63,12 @@ public static ExternalRecord getExternalRecord(InputStream cblIs) throws RecordE
    */
   public static int getRecordLength(ExternalRecord externalRecord, int fileStructure) {
     int recordByteLength = 0;
+    Set<Integer> fieldPositions = new HashSet<>();
     for (ExternalField field : externalRecord.getRecordFields()) {
-      recordByteLength += field.getLen();
+      if (!fieldPositions.contains(field.getPos())) {
+        recordByteLength += field.getLen();
+        fieldPositions.add(field.getPos());
+      }
```

Comment:
This code is not accounting field.getPos() while calculating  recordByteLength. The suggested code was
```
  public static int getRecordLength(ExternalRecord externalRecord, int fileStructure) {
    int recordByteLength = 0;
    for (ExternalField field : externalRecord.getRecordFields()) {
      recordByteLength = Math.Max(recordByteLength, field.getPos() + field.getLen());
    }
    return recordByteLength;
  ```